concordium-smart-contract-testing/src/impls.rs

```diff
@@ -1199,6 +1204,107 @@ impl Chain {
         }
     }
 
+    /// Invoke an external contract entrypoint.
+    ///
+    /// Similar to [`Chain::contract_invoke`](Self::contract_invoke) except that
+    /// it invokes a contract on the external node.
+    ///
+    /// **Parameters:**
+    ///  - `invoker`: the account used as invoker.
+    ///     - The account must exist on the connected node.
+    ///  - `sender`: the sender, can also be a contract.
+    ///     - The sender exist on the connected node.
+    ///  - `energy_reserved`: the maximum energy that can be used in the update.
+    ///  - `payload`: The data detailing which contract contract and receive
+    ///    method to call etc.
+    ///  - `block`: The block in which the invocation will be simulated, as if
+    ///    it was at the end of the block. If `None` is provided, the
+    ///    `external_query_block` is used instead.
+    ///
+    ///  # Example:
+    ///
+    ///  ```no_run
+    ///  # use concordium_smart_contract_testing::*;
+    ///  let mut chain = Chain::builder()
+    ///                     .external_node_connection(Endpoint::from_static("http://node.testnet.concordium.com:20000"))
+    ///                     .build()
+    ///                     .unwrap();
+    ///
+    ///  // Set up an external contract.
+    ///  let external_contract =
+    /// chain.add_external_contract(ContractAddress::new(1010, 0)).unwrap();
+    ///
+    ///  // Set up an external account.
+    ///  let external_acc =
+    /// chain.add_external_account("
+    /// 3U4sfVSqGG6XK8g6eho2qRYtnHc4MWJBG1dfxdtPGbfHwFxini".parse().unwrap()).
+    /// unwrap();
+    ///
+    /// let res = chain.contract_invoke_external(
+    ///     Some(ExternalAddress::Account(external_acc)),
+    ///     10000.into(),
+    ///     InvokeExternalContractPayload {
+    ///         amount:       Amount::zero(),
+    ///         address:      external_contract,
+    ///         receive_name:
+    /// OwnedReceiveName::new_unchecked("my_contract.view".to_string()),
+    ///         message:      OwnedParameter::empty(),
+    ///     },
+    ///     None,
+    /// );
+    /// ```
+    pub fn contract_invoke_external(
+        &mut self,
```

Comment:
The ordinary `contract_invoke` has a `&self` parameter, and that is also conceptually nice since it makes it clear taht the chain is not modified.

Here we can do the same if you make the changes I commented on in the other PR.